### PR TITLE
LPS-97094 User's Personal Data is not fully anonymized in Related Assets

### PR DESCRIPTION
**Pull request by stsquared99**

### CODE REVIEW

File: modules/util/portal-tools-service-builder/src/main/resources/com/liferay/portal/tools/service/builder/dependencies/base_uad_anonymizer.ftl [hunk 1]
```diff
@@ -38,6 +43,17 @@ public abstract class Base${entity.name}UADAnonymizer extends DynamicQueryUADAno
...
+						<#if hasAssetEntry && stringUtil.equals(uadUserIdEntityColumn.name, "userId")>
+							AssetEntry assetEntry = assetEntryLocalService.fetchEntry(${entity.name}.class.getName(), ${entity.varName}.get${entity.getPKMethodName()}());
+
+							if (assetEntry != null) {
+								assetEntry.setUserId(anonymousUser.getUserId());
+								assetEntry.setUserName(anonymousUser.getFullName());
+
+								assetEntryLocalService.updateAssetEntry(assetEntry);
+							}
+						</#if>
```
Comment (pei-jung) [inline modules/util/portal-tools-service-builder/src/main/resources/com/liferay/portal/tools/service/builder/dependencies/base_uad_anonymizer.ftl]: We can extract this to a separate method `protected updateAsset(Object entity, long userId, User anonyUser)` so it'll be easier to override if an entity requires special logic to update its assets.

Actually, we can break it down even further to have another method for getting the asset entry `protected getAssetEntry(Object entity)`. So for cases like the WikiPage, we only need to override this simple method.

**Comment by pei-jung** [comment]
Hi @adolfopa,

Here is our attempt to update the asset generically. The reason we didn't go with that approach is because we see that every models handles its assets differently. For example, DL and MB entities update only the asset entries, whereas bookmarks, wiki and blogs entities update both asset entries and links. I'm not very familiar with the assets framework so I'm not sure if doing this in a generic way will be appropriate and if doing a partial update (ie. update only the asset entry but not the links for wiki pages) will cause any harm. I'm okay with having a generic logic for updating the assets, and each application provide additional logic when necessary. Can you take a look at these changes and let us know what you think? Thanks!

**Comment by adolfopa** [comment]
Looks good to me.  I also think it is fine to cover only the basic case and override the method when additional logic is required.

Thanks a lot!

**Comment by pei-jung** [comment]
@adolfopa thanks for taking a look!

@stsquared99 can you update the pull and resend? Thanks!

**Pull request closed without merging**